### ISSUE TITLE
Lecture nine - Classification PCA

### ISSUE THREAD

**Issue by jimjinyan** [jimjinyan/Go-Just-Go#18]
Start with a data and then asign a label to the data!
-  below will focus on image classification

Problems for Calssification(image)
- Viewpoints, lunmination condition, Scale, deformation, Occlusion, ......

Big breakthroug for image classification due to
-  machine learning (data driven) by big data
    - need to people label on this data as training data

General Process for Machine Learning(Supervised):
- Input a training set of N datapoints, each labeled with one of K different calsses
- Machine learn the lable of every one of the training set
    - Algorithm: kNN, SVM, Decision Trees, Random Forests, Bagging, Boosting, etc.....
- Predict labels for a test set of data and compare the true lables to the ones prediced by the calssifier 

Unsupervised learning:
- Find patterns in the data wihtout labels
- Algorithm: PCA, MDS, Clustering
- Sometimes we do not know the label, we do not have training data or too many data to classify by people, so we need unsupervised learning to learn data first

- Classification is based on the **feature** space. 
   - So it is import to select the fetures carefully!
   - Feature selection is one of main tasks in Machine Learning and can be done by cross-validation
 
Now look the machine learning from kNN algorithm
- Predict class of new data point by majority vote of K nearest neighbors
- The size of beighborhood is matter! (since it decide how many neighbors are going to vote)
    - it is what the k means
        - if the k is 1, then it is equal to Voronoi diagram

1 NN properties
-  simple and good for low dimensional
- Rough decision boundary and may have 'island'
- Training complex for N data points--- order1
- Testng complexity for M data points---  order is M*N
    -  so basically no work for training and all wrok done by testing(not good takes time for usage)
- Error on training - 0
- Variance: very high!; Bias: is low
    - to lower variance--increasing the number of k (but bias will be higher)
        - then how to choice the k?
           - by cross validation to minimize the loss fucntion
                - then need to decide which loss fucntion should be!
- Voronoi diagram
    - basically it means 每个V多边形内点到该生成元距离短于到其它生成元距离
    - [refference](https://blog.csdn.net/kevin_darkelf/article/details/81455687)
- Cross-validation
    -  see lab4
    - 参数（parameters）：指的是模型可以根据所输入的数据自动学习出来的变量。常见的模型参数有神经网络中的权重、支持向量机中的支持向量等。
    - 超参数（hyper-parameters）：超参数的值不是通过算法学出来的，而是人通过经验设定的（当然，也可以设计一个学习算法来找出模型最优的超参数，比如使用网格法搜索SVM最优的c和gamma）。深度学习中常见的模型参数有学习率、网络层数、网络层节点数等。
    - **The differnce between validation data and testing data**

How a image change to data
    - Each image is a vector whic has red, green and blue, so just like unroll the image 32X32X3 for example

Image Comparision:
    - just take the difference between image vectors
    - Distance Metrics
        -  L1, L2 or Lp ...

But the most pixes are backgroud pixes, so the pefrmance is not good!(example used in class)
    - So take the different **features**
        - now the features are pixes the best one is Artificial Neural Network
    - **Feature selection is the most important task!**
        -  this is way there are so many sense for auto drive car!
   -  Then, causing the problems of **Curse of Dimensionality**
        -  Then the available data becomes spares

Dimensionality Reduction
    - need to perserve the distance
    - **basic idea:** project the high dimnsional data onto a lower dimensional subspace that best "fits" the data.
        -  most of them are linear models
        -  how to decide its accuracy and how many dimension we should use--PCA

Principal Components Analysis(PCA)
- find a direction of maximum variance / minimize orthogonal distances
- The diffrence between PCA and linear Regression
    - Linear Regaression is along his original axis
    - PCA is orthogonal projects (rotation of the original axis)

PCA Algorithm
    - TBD (will place a new issue for this topic)
    - 由于各变量之间存在一定的相关关系，因此可以考虑将关系紧密的变量变成尽可能少的新变量，使这些新变量是两两不相关的，那么就可以用较少的综合指标分别代表存在于各个变量中的各类信息。主成分分析与因子分析就属于这类降维算法
  
How many principal Components? 
   -  Decide by Screeplot

[A very good refference: 主成分分析（PCA）原理详解](https://zhuanlan.zhihu.com/p/37777074)

Multidimensional Scaling
Not the data but the distance for pairwise points
- get the distance matrix
- some linear algebra calculation(no details, need to check)
- Perfrom PCA on this matrix

Nonlinear Methods
- no details and need to check

**Comment by jimjinyan** [comment]
Close with three open issues. One will be seprated and two will check later when met problems or solve later when have time.
